antsibull/build_ansible_commands.py

```diff
@@ -86,10 +105,16 @@ def write_manifest(package_dir, release_notes: t.Optional[ReleaseNotes] = None,
         f.write('recursive-include ansible_collections/ **\n')
 
 
-def write_setup(ansible_version, ansible_base_version, collection_deps, package_dir):
+def write_setup(ansible_version: PypiVer,
+                ansible_base_version: PypiVer,
+                collection_deps: str,
+                package_dir: str) -> None:
     setup_filename = os.path.join(package_dir, 'setup.py')
 
-    setup_tmpl = Template(pkgutil.get_data('antsibull.data', 'ansible-setup_py.j2').decode('utf-8'))
+    template = pkgutil.get_data('antsibull.data', 'ansible-setup_py.j2')
+    if template is None:
+        raise Exception('Cannot find ansible-setup_py.j2 in antsibull.data package')
+    setup_tmpl = Template(template.decode('utf-8'))
```

Comment:
Is there a reason not to just consider this a broken install and traceback?  It's kind of like if `import antsibull.ansible_base` were to fail.  We don't guard against all of those.

Reply:
The thing is that pyre complains, because `template` can be `None`. I'm not sure whether it would accept `assert templace is not None` instead, but that's not much better than explicitly raising an exception.

Reply:
Hmm... so bad API that we don't control :-(.... (pkgutil.get_data should probably throw a different exception in the place where it currently returns None [if the underlying loader doesn't support get_data]).

I did check and `assert template is not None` does work.

So it feels like this to me... `pkgutil.get_data()` will only return None if the loader for antsibull.data does not support get_data.  That's likely to be the case only in cases that we control (if we change antsibull.data to be a namespace package, for instance).  Given that we're in control, I lean towards using `assert None` to tell type checkers that we know the return value will never be None in our case.  However, if we don't trust the stdlib not to reimplement and return None in a different circumstance, or we think we could change our package so that it uses a different loader in the future, then checking for None and raising an Exception seems right.  If I'm reading the pkgutil docs correctly, though, I think we'd have to change our code regardless in that circumstance. We can't future proof it; if we ever throw that exception we'll have to change the code to fix it.

If we do the latter, maybe we need to wrap pkgutil.get_data in a function which detects None and throws an Exception?  I'm not sure what the exception type should be... from the documentation, I think that maybe RuntimeError is correct (it's a RuntimeError that a loader which doesn't implement get_data is being used)?

Reply:
I added an utility function handling this in 372c308c80e027db5f3f7ab5003d21d5492669ec.